### ISSUE TITLE
Speakercraft

### ISSUE THREAD

**Issue by BarryHA666** [mattsaxon/HASS-SpeakercraftMediaPlayer#50]
Hi,

I'm a HA newby, and some simple question.

I'm using your great speakercraft integration which works great on my MZC-66. Using an Nport for the distance. I made some cards in my gui but i can only switch the channels on, and not off. Do you have some examples how to make one button to turn on or off on channel?

Thanks in advance.

Barry.

**Comment by mattsaxon** [comment]
I just use the standard media controls, they have most of the relevant things, both on and off, plus volume and mute. How have you done your existing cards, please post the yaml.

**Comment by BarryHA666** [comment]
Thanks for the reaction. I used a standard card configuration and selected the entity media_player.badkamer. But after some more research i found out that i don't get an update back. Maybe it is some setting in the Nport (5110A). Could that be the issue?

**Comment by BarryHA666** [comment]
I think i should get an update back from the amp when i turn it on using HA, right?

**Comment by BarryHA666** [comment]
source_list:
  - Chromecast 1
  - Chromecast 2
  - Chromecast 3
  - Chromecast 4
Bass: '0'
Treble: '0'
Party Mode: false
Party Master: false
Volume DB: 0
device_class: speaker
icon: mdi:speaker
friendly_name: Zolder
supported_features: 3468

This is the yaml from the button using entity media_player.zolder

Barry.

**Comment by mattsaxon** [comment]
Yes, that would be my guess. I imagine the messages are not coming back. Turn on debug logging and take a trace, then post the relevant part of the file. Please post as an attachment rather than online text

**Comment by BarryHA666** [comment]
Hi Matt,

Hereby the attachment with the log. I restarted HA and switched some input to on. Today i downloaded the EZ tools from Speakercraft and configured the amp again, just to be sure that everything works, I also can see the model and firmware, wich is the 2.2.8. To connect to the amp i used the same setup I use with HomeAssistant. The only thing i have to change is the Nport setting. To connect to the amp using the EZ Tools i have to switch to RealCom mode in the Nport. To use it with HA i switch the mode to TCP Server. 
[home-assistant_2023-01-31T20-36-11.482Z.log](https://github.com/mattsaxon/HASS-SpeakercraftMediaPlayer/files/10550927/home-assistant_2023-01-31T20-36-11.482Z.log)
 
Hope you can find something here. Let me know if you need more information, or i have to change something.

Thanks for looking!

Barry.

**Comment by mattsaxon** [comment]
OK, I see the log and you are not getting messages sent back, in my log I get extra data such as 

`2023-01-31 21:20:02.731 DEBUG (MainThread) [custom_components.speakercraft_media.speakercraft_media] status message 550b2000000e014b00000f`

After setting the source, these are the messages that update the front end.

When you say you are changing NPort settings, what settings are you changing and why? I don't have to change my setting when swapping from EZ tools to HA control.

**Comment by BarryHA666** [comment]
Hi,

It's working now! Did some tests with the Nport, and now it is working! Another question, is there a possibility to select a default source for every zone? Right now, when a zone is turned on, it has source 1, but it would be nice to have zone 1 with default source 1, and zone 2 default source 2 and so on.

Is this possible? If so how does the yaml look like?

Thanks for helping me out!

Barry

**Comment by mattsaxon** [comment]
that's not implemented in the integration.

I'd recommend you achieve this using EZ tools that way it will apply even when you use manual controls.

Alternatively you can automate this with HA, set an automation to trigger when the device is turned on

**Comment by BarryHA666** [comment]
Thanks! I will try and use NodeRed to get it further to work! 

Greets!

**Comment by sjeffrey101** [comment]
I dont believe it will apply when its done in EZ-tools.  For some reason none of the automations apply when its controlled via serial.

nodered or HA automations are the way to go.  I'd probably remove the default source line in the config, so then it just sends the turn on command, and it would turn onto the source it was last on,  then the automation can run.  This way probably most the time it would already be on that source any way.

**Comment by BarryHA666** [comment]
Thanks guys! And i made it already the way yoy explained. Most of the times the settings are correct.Very nice implementation!Barry.Op 2 feb. 2023 09:51 schreef sjeffrey101 ***@***.***>:
I dont believe it will apply when its done in EZ-tools.  For some reason none of the automations apply when its controlled via serial.
nodered or HA automations are the way to go.  I'd probably remove the default source line in the config, so then it just sends the turn on command, and it would turn onto the source it was last on,  then the automation can run.  This way probably most the time it would already be on that source any way.
